src/src/ArrayView.hpp

```diff
@@ -246,64 +238,57 @@ class ArrayView
    * only difference between the types is a const specifier.
    */
   inline LVARRAY_HOST_DEVICE CONSTEXPRFUNC
-  operator ArraySlice<T, NDIM, INDEX_TYPE>() const noexcept
+  operator ArraySlice<T, NDIM, UNIT_STRIDE_DIM, INDEX_TYPE>() const noexcept
   {
-    return ArraySlice<T, NDIM, INDEX_TYPE>( m_data, m_dims, m_strides );
+    return ArraySlice<T, NDIM, UNIT_STRIDE_DIM, INDEX_TYPE>( m_data, m_dims, m_strides );
   }
 
   template< typename U = T >
   inline LVARRAY_HOST_DEVICE CONSTEXPRFUNC
   operator typename std::enable_if< !std::is_const<U>::value,
-                                    ArraySlice<T const, NDIM, INDEX_TYPE> const >::type
+                                    ArraySlice<T const, NDIM, UNIT_STRIDE_DIM, INDEX_TYPE> const >::type
     () const noexcept
   {
-    return ArraySlice<T const, NDIM, INDEX_TYPE>( m_data, m_dims, m_strides );
+    return ArraySlice<T const, NDIM, UNIT_STRIDE_DIM, INDEX_TYPE>( m_data, m_dims, m_strides );
+  }
+
+  inline LVARRAY_HOST_DEVICE CONSTEXPRFUNC
+  ArraySlice<T, NDIM, UNIT_STRIDE_DIM, INDEX_TYPE> toSlice() const noexcept
+  {
+    return ArraySlice<T, NDIM, UNIT_STRIDE_DIM, INDEX_TYPE>( m_data, m_dims, m_strides );
+  }
+
+  template< typename U = T >
+  inline LVARRAY_HOST_DEVICE CONSTEXPRFUNC
+  ArraySlice<T const, NDIM, UNIT_STRIDE_DIM, INDEX_TYPE> toSliceConst() const noexcept
+  {
+    return ArraySlice<T const, NDIM, UNIT_STRIDE_DIM, INDEX_TYPE>( m_data, m_dims, m_strides );
   }
 
 
   /**
    * User defined conversion to convert a ArraySlice<T,1,NDIM> to a raw pointer.
    * @return a copy of m_data.
    */
-  template< int U=NDIM >
+  template< int _NDIM=NDIM, int _UNIT_STRIDE_DIM=UNIT_STRIDE_DIM >
   LVARRAY_HOST_DEVICE CONSTEXPRFUNC inline operator
-  typename std::enable_if< U == 1, T * const restrict >::type () const noexcept restrict_this
+  typename std::enable_if< _NDIM == 1 && _UNIT_STRIDE_DIM == 0, T * const restrict >::type () const noexcept restrict_this
   {
     return m_data;
   }
 
-  /**
-   * Function to convert to a reduced dimension array. For example, converting from
-   * a 2d array to a 1d array is valid if the last dimension of the 2d array is 1.
-   */
-  template< int U=NDIM >
-  inline CONSTEXPRFUNC
-  typename std::enable_if< (U > 1), ArrayView<T, NDIM - 1, INDEX_TYPE, DATA_VECTOR_TYPE> >::type
-  dimReduce() const noexcept
```

Comment:
Is this functionally somewhere else now?

Reply:
We weren't actually using it anywhere so I removed it.

Reply:
We really should use this for single point cell centered data. The use case is specifically that we use 2d arrays to store material data (numElems,numQuadPoints), but we using cell centered quadrature, only a 1d array is required. The dimReduce allows us to just treat the 2d array like a 1d array and skip "dim 1" index.